core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.spec.ts

```diff
@@ -178,12 +178,16 @@ describe('ExplorationFooterComponent', () => {
 
   it('should not show hints after user finishes practice session' +
   ' and results are loaded.', () => {
+    const explorationId = 'exp1';
+    spyOn(contextService, 'getExplorationId').and.returnValue(explorationId);
     spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(true);
     expect(component.hintsAndSolutionsAreSupported).toBeTrue();
 
     spyOnProperty(questionPlayerStateService, 'resultsPageIsLoadedEventEmitter')
       .and.returnValue(mockResultsLoadedEventEmitter);
 
+    component.explorationId = explorationId;
```

Comment:
Isn't `component.explorationId` assigned a value here:
https://github.com/oppia/oppia/blob/516594824a756f0d6b7716e1f07f83bb420d15a2/core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.ts#L103

Why are we assigning a value before `component.ngOnInit();` is executed?

Reply:
I agree, this seems redundant since we are mocking `contextService.getExplorationId` in line 182 in this file. Here and below, it should be sufficient to mock getExplorationId.

Reply:
Initially I had not added assigned the value of explorationId before `component.ngOnInit();` but it was resulting in flakes, and tests sometimes passing and sometimes failing, I related it with this comment:
https://github.com/oppia/oppia/blob/516594824a756f0d6b7716e1f07f83bb420d15a2/core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.ts#L94-L103

and to deal with this I specifically assigned the value of exploration Id before calling `component.ngOnInit();` so that in case the try fails and the explorationId is unavailable, use preassigned value of explorationId.

But I have impemented it in a seperate way now, so the exploration Id is only used if it's not the practice session type. That also works. PTAL. Thanks!